compare_py_and_rs_json.py

```diff
@@ -0,0 +1,188 @@
+    "Integer",
+    "Index",
+    "Size",
+    "NonZeroInteger",
+    "Version",
+    "Option<Integer>",
+    "Option<Version>",
+    "Option<Size>",
+}
+# Uncomment this to show different integer errors
+INTEGER_ALIASES = {}
+
+
+def strip_comment(txt: str) -> str:
+    return "\n".join(x for x in txt.splitlines() if not x.strip().startswith("//"))
+
+
+def compare_file(item_name: str, file1: Path, file2: Path) -> bool:
+    api1 = json.loads(strip_comment(file1.read_text()))
+    api2 = json.loads(strip_comment(file2.read_text()))
+    assert len(api1) > 0
+    assert len(api2) > 0
+
+    ok = True
+    if len(api1) != len(api2):
+        print(f"{item_name}: number of items differs {len(api1)!r} vs {len(api2)!r}")
+        ok = False
+
+    api1 = api1[0]
+    api2 = api2[0]
+
+    if api1["label"] != api2["label"]:
+        print(f"{item_name}: label differs {api1['label']!r} vs {api2['label']!r}")
+        ok = False
+
+    # Check req
+    req1 = api1["req"]
+    req2 = api2["req"]
+    if req1["cmd"] != req2["cmd"]:
+        print(f"{item_name}: req cmd differs {req1['cmd']!r} vs {req2['cmd']!r}")
+        ok = False
+
+    if req1.get("unit") != req2.get("unit"):
+        print(f"{item_name}: req unit differs {req1.get('cmd')!r} vs {req2.get('cmd')!r}")
+        ok = False
+
+    req1_fields = {x["name"]: x for x in req1["other_fields"]}
+    req2_fields = {x["name"]: x for x in req2["other_fields"]}
+    if req1_fields.keys() ^ req2_fields.keys():
+        print(
+            f"{item_name}: different other_fields in req: {req1_fields.keys() ^ req2_fields.keys()!r}"
+        )
+        ok = False
+    for field in req1_fields.keys() & req2_fields.keys():
+        t1 = req1_fields[field]["type"]
+        t2 = req2_fields[field]["type"]
+        if t1 != t2 and not (t1 in INTEGER_ALIASES and t2 in INTEGER_ALIASES):
+            print(f"{item_name}: req field {field!r} type differs {t1!r} vs {t2!r}")
+            ok = False
+
+    def _compare_fields(item_name: str, fields1: List[dict], fields2: List[dict]) -> bool:
+        ok = True
+        fields1 = {x["name"]: x for x in fields1}
+        fields2 = {x["name"]: x for x in fields2}
+        if fields1.keys() ^ fields2.keys():
+            print(f"{item_name}: different fields: {fields1.keys() ^ fields2.keys()!r}")
+            ok = False
+        for field in fields1.keys() & fields2.keys():
+            t1 = fields1[field]["type"]
+            t2 = fields2[field]["type"]
+            if t1 != t2 and not (t1 in INTEGER_ALIASES and t2 in INTEGER_ALIASES):
+                print(f"{item_name}: field {field!r} type differs {t1!r} vs {t2!r}")
+                ok = False
+        return ok
+
+    # Check reps (only status can be checked)
+
+    rep1_ok = next(x for x in api1["reps"] if x["status"] == "ok")
+    rep2_ok = next(x for x in api2["reps"] if x["status"] == "ok")
+    ok &= _compare_fields(
+        f"{item_name} ok rep fields", rep1_ok["other_fields"], rep2_ok["other_fields"]
```

Comment:
```suggestion
        f"{item_name}: ok rep fields", rep1_ok["other_fields"], rep2_ok["other_fields"]
```
Keep the standard format that you've used all along